tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,898 @@
+
+
+class WithEstimates(WithTradingSessions, WithAssetFinder):
+    """
+    ZiplineTestCase mixin providing cls.loader and cls.events as class
+    level fixtures.
+
+
+    Methods
+    -------
+    make_loader(events, columns) -> PipelineLoader
+        Method which returns the loader to be used throughout tests.
+
+        events : pd.DataFrame
+            The raw events to be used as input to the pipeline loader.
+        columns : dict[str -> str]
+            The dictionary mapping the names of BoundColumns to the
+            associated column name in the events DataFrame.
+    """
+
+    # Short window defined in order for test to run faster.
+    START_DATE = pd.Timestamp('2014-12-28')
+    END_DATE = pd.Timestamp('2015-02-03')
+
+    @classmethod
+    def make_loader(cls, events, columns):
+        raise NotImplementedError('make_loader')
+
+    @classmethod
+    def init_class_fixtures(cls):
+        cls.sids = cls.events[SID_FIELD_NAME].unique()
+        cls.columns = {
+            Estimates.event_date: 'event_date',
+            Estimates.fiscal_quarter: 'fiscal_quarter',
+            Estimates.fiscal_year: 'fiscal_year',
+            Estimates.estimate: 'estimate'
+        }
+        cls.loader = cls.make_loader(cls.events, {column.name: val for
+                                                  column, val in
+                                                  cls.columns.items()})
+        cls.ASSET_FINDER_EQUITY_SIDS = list(
+            cls.events[SID_FIELD_NAME].unique()
+        )
+        cls.ASSET_FINDER_EQUITY_SYMBOLS = [
+            's' + str(n) for n in cls.ASSET_FINDER_EQUITY_SIDS
+        ]
+        super(WithEstimates, cls).init_class_fixtures()
+
+
+class WithWrongNumQuarters(WithEstimates):
+    """
+    ZiplineTestCase mixin providing cls.events as a class level fixture and
+    defining a test for all inheritors to use.
+
+    Attributes
+    ----------
+    events : pd.DataFrame
+        A simple DataFrame with columns needed for estimates and a single sid
+        and no other data.
+
+    Tests
+    ------
+    test_wrong_num_quarters_passed()
+        Tests that loading with an incorrect quarter number raises an error.
+    """
+    events = pd.DataFrame({SID_FIELD_NAME: 0},
+                          columns=[SID_FIELD_NAME,
+                                   TS_FIELD_NAME,
+                                   EVENT_DATE_FIELD_NAME,
+                                   FISCAL_QUARTER_FIELD_NAME,
+                                   FISCAL_YEAR_FIELD_NAME,
+                                   'estimate'],
+                          index=[0])
+
+    def test_wrong_num_quarters_passed(self):
+        dataset = QuartersEstimates(-1)
+        engine = SimplePipelineEngine(
+            lambda x: self.loader,
+            self.trading_days,
+            self.asset_finder,
+        )
+        p = Pipeline({c.name: c.latest for c in dataset.columns})
+
+        with self.assertRaises(ValueError):
```

Comment:
since this is only a valueerror could we check the message? Also, there is some logic around the formatting that we want to test.